src/posix/platform/resolver.cpp

```diff
@@ -298,27 +335,18 @@ void Resolver::Process(const otSysMainloopContext &aContext)
 
 void Resolver::SetUpstreamDnsServers(const otIp6Address *aUpstreamDnsServers, int aNumServers)
 {
-    mUpstreamDnsServerCount = 0;
+    mUpstreamDnsServerCount = OT_MIN(aNumServers, static_cast<int>(kMaxUpstreamServerCount));
```

Comment:
Should we maybe change `int aNumServers` to be `uint32_t` or `uint16`? Having it be `int` does not make sense as we don't expect it to be negative?